.github/actions/setup-binary/action.yml

```diff
@@ -0,0 +1,135 @@
+# Copyright 2023 The Authors (see AUTHORS file)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: 'Setup Binary'
+
+description: |-
+  Use this action to install and cache a binary for use within a GitHub workflow.
+
+inputs:
+  name:
+    description: 'The name of the binary.'
+    required: true
+  download_url:
+    description: 'The download URL for the binary.'
+    required: true
+  install_dir:
+    description: 'The directory to install the binary (e.g. install/dir).'
+    default: '.'
+    required: false
+  binary_dir:
+    description: 'The directory to the binary. This is useful when the binary is located in a sub-path of the downloaded archive (e.g. install/dir/sub/path).'
+    required: false
+  enable_cache:
+    description: 'Enable the use of caching for the binary.'
+    default: 'false'
+    required: false
+  override_cache_key:
+    description: 'Override the default value of the cache key.'
+    required: false
```

Comment:
I think we could unify these into a single value, `cache_key`. In the special case that `cache_key` is the literal string "none" or false, do not cache.

Reply:
Went with empty string, do not cache.